src/api/utils/api_utils.py

```diff
@@ -98,3 +108,283 @@ def create_media_file(file, name):
+                (Q(audience=FeatureFlagConstants().for_all_except()) & ~Q(communities=community))
+            ).exclude(expires_on__lt=datetime.now()).prefetch_related('communities')
+    
+    return feature_flags
+def get_actions_for_community(community, context):
+    actions = Action.objects.select_related('image', 'community').prefetch_related('tags', 'vendors').filter(community=community)
+    if not context.is_sandbox:
+        if context.user_is_logged_in and not context.user_is_admin():
+            actions = actions.filter(Q(user__id=context.user_id) | Q(is_published=True))
+        else:
+            actions = actions.filter(is_published=True)
+    return actions.distinct()
+
+
+def get_events_for_community(community, context):
+    today = datetime.now()
+    shared_months = 2
+    hosted_months = 6
+    earliest_shared = today - timedelta(weeks=4 * shared_months)
+    earliest_hosted = today - timedelta(weeks=4 * hosted_months)
+    
+    events = Event.objects.select_related('image', 'community').prefetch_related('tags', 'invited_communities').filter(
+        community__id=community.id, start_date_and_time__gte=earliest_hosted)
+    
+    shared = community.events_from_others.filter(is_published=True, start_date_and_time__gte=earliest_shared)
+    if not context.is_sandbox and events:
+        if context.user_is_logged_in and not context.user_is_admin():
+            events = events.filter(Q(user__id=context.user_id) | Q(is_published=True))
+        else:
+            events = events.filter(is_published=True)
+    all_events = [*events, *shared]
+    
+    return all_events
+
+
+def get_teams_for_community(community):
+    pass
+
+def get_testimonials_for_community(community, context):
+    
+    testimonials = Testimonial.objects.filter(
+        community=community, is_deleted=False).prefetch_related('tags__tag_collection', 'action__tags', 'vendor',
+                                                                'community')
+    
+    if not context.is_sandbox:
+        if context.user_is_logged_in and not context.user_is_admin():
+            testimonials = testimonials.filter(Q(user__id=context.user_id) | Q(is_published=True))
+        else:
+            testimonials = testimonials.filter(is_published=True)
+        
+    return testimonials.distinct()
+
+def get_vendors_for_community(community, context):
+    vendors = community.community_vendors.filter(is_deleted=False)
+    
+    if not context.is_sandbox:
+        if context.user_is_logged_in and not context.user_is_admin():
+            vendors = vendors.filter(Q(user__id=context.user_id) | Q(is_published=True))
+        else:
+            vendors = vendors.filter(is_published=True)
+    return vendors.distinct()
+
+def get_tags_collections():
+    return TagCollection.objects.filter(is_deleted=False)
+
+# .......................... site setup utils ..........................
+def load_homepage_data(context, args, community, id, page_settings):
```

Comment:
These methods aren't used?  If not remove

Reply:
They might be required later

Reply:
remove and put in a different file or branch so that we don't clutter this PR

Reply:
ok